rasa/shared/core/domain.py

```diff
@@ -1466,84 +1487,6 @@ def get_responses_with_multilines(
-        Returns:
-            The intent properties as they are used in domain files.
-        """
-        intent_properties = copy.deepcopy(self.intent_properties)
-        sorted_intent_properties = sorted(intent_properties.items())
-        intents_for_file = []
-
-        for intent_name, intent_props in sorted_intent_properties:
-            if (
-                intent_name in rasa.shared.core.constants.DEFAULT_INTENTS
-                and intent_name not in self.overridden_default_intents
-            ):
-                # Default intents should be not dumped with the domain
-                continue
-            # `use_entities` and `ignore_entities` in the domain file do not consider
-            # the role and group labels remove them from the list to make sure to not
-            # put them into the domain file
-            use_entities = set(
-                entity
-                for entity in intent_props[USED_ENTITIES_KEY]
-                if rasa.shared.core.constants.ENTITY_LABEL_SEPARATOR not in entity
-            )
-            ignore_entities = set(self.entities) - use_entities
-            if len(use_entities) == len(self.entities):
-                intent_props[USE_ENTITIES_KEY] = True
-            elif len(use_entities) <= len(self.entities) / 2:
-                entities = list(use_entities)
-                entities.sort()
-                intent_props[USE_ENTITIES_KEY] = entities
-            else:
-                entities = list(ignore_entities)
-                entities.sort()
-                intent_props[IGNORE_ENTITIES_KEY] = entities
-            intent_props.pop(USED_ENTITIES_KEY)
-            intents_for_file.append({intent_name: intent_props})
-
-        return intents_for_file
-
-    def _transform_entities_for_file(self) -> List[Union[Text, Dict[Text, Any]]]:
-        """Transform entity properties for displaying or writing to a domain file.
-
-        Returns:
-            The entity properties as they are used in domain files.
-        """
-        entities_for_file: List[Union[Text, Dict[Text, Any]]] = []
-
-        for entity in self.entities:
-            if entity in self.roles and entity in self.groups:
-                entities_for_file.append(
-                    {
-                        entity: {
-                            ENTITY_GROUPS_KEY: self.groups[entity],
-                            ENTITY_ROLES_KEY: self.roles[entity],
-                        }
-                    }
-                )
-            elif entity in self.roles:
-                entities_for_file.append(
-                    {entity: {ENTITY_ROLES_KEY: self.roles[entity]}}
-                )
-            elif entity in self.groups:
-                entities_for_file.append(
-                    {entity: {ENTITY_GROUPS_KEY: self.groups[entity]}}
-                )
-            else:
-                entities_for_file.append(entity)
-
-        return entities_for_file
-
     def cleaned_domain(self) -> Dict[Text, Any]:
```

Comment:
Probably not 🤔 It appears useful only insofar as default keys get removed from the domain. I'll look now to remove it.

Reply:
Having had a look at this, `cleaned_domain` is useful mainly for removing the defaults that get added, e.g. `e2e_actions: [ ]` etc. I can reduce the scope of this function only to removing keys that have empty values, would that be ok?

Reply:
Do we need the defaults in `self._data` in the first place?

Reply:
Yes, they get added because of those chained importers requiring the default values for certain keys.

Reply:
Aha, think I got to the bottom of why I actually had to add all the extra keys, they were added during the merging process and `rasa train` unit tests would fail when comparing the metadata.domain with the loaded domain from file. Making some amendments now to sort this out now 😄